### PR TITLE
Implement simple vmemcache linear allocator

### PR DESCRIPTION
**Pull request by ldorau**
This change is [Reviewable](https://reviewable.io/reviews/pmem/vmemcache/5)

### CODE REVIEW

**Comment by ldorau** [comment]
This PR contains PR https://github.com/pmem/vmemcache/pull/4 yet.

File: src/vmemcache_heap.c [hunk 1]
```diff
@@ -46,7 +52,26 @@ vmcache_heap_create(void *addr, size_t size, size_t fragment_size)
...
+	struct heap *heap;
+
+	heap = Zalloc(sizeof(struct heap));
+	if (heap == NULL) {
+		ERR("!Zalloc");
+		return NULL;
+	}
+
+	util_mutex_init(&heap->lock);
+	util_mutex_lock(&heap->lock);
```
Comment (pbalcer) [inline src/vmemcache_heap.c]: you don't need to grab this lock immediately.

File: src/vmemcache_heap.c [hunk 2]
```diff
@@ -56,6 +81,15 @@ void
 vmcache_heap_destroy(struct heap *heap)
 {
 	LOG(3, "heap %p", heap);
+
+	util_mutex_lock(&heap->lock);
```
Comment (pbalcer) [inline src/vmemcache_heap.c]: same, we don't need to grab this lock

**Comment by marcinslusarz** [comment]
simply -> simple

**Pull request merged**